BraintreeCore/src/main/java/com/braintreepayments/api/ShopperInsightsRequest.kt

```diff
@@ -0,0 +1,17 @@
+package com.braintreepayments.api
+
+/**
+ * Data class representing a request for shopper insights.
+ *
+ * Note: **This feature is in beta. It's public API may change in future releases.**
+ */
+data class ShopperInsightsRequest(
+    val email: String?,
+    val phone: BuyerPhone?
+) {
+    init {
+        require(email != null || phone != null) {
```

Comment:
@tdchow @sarahkoop placing the validation inline.

Reply:
I think we want to call back an error later in the flow if one of these is null (in whatever method they pass this request object into) - the merchant will already have a callback setup to handle errors for that method, so it's no extra implementation lift on their end. If we throw an error here, they have to have a separate check